tcslackbuildnotifier-core/src/main/java/slacknotifications/SlackNotificationImpl.java

```diff
@@ -239,50 +241,100 @@ private void postViaWebHook() throws IOException {
             Loggers.SERVER.info("SlackNotificationListener :: Preparing message for URL " + url);
 
             WebHookPayload requestBody = new WebHookPayload();
-            
-            requestBody.setChannel(this.getChannel());
-            requestBody.setUsername(this.getBotName());
-            requestBody.setIcon_url(this.getIconUrl());
 
-            HttpPost httppost = new HttpPost(url);
+            List<Commit> commits = this.payload.getCommits();
 
-            if (this.payload != null) {
-                requestBody.setText(payload.getBuildDescriptionWithLinkSyntax());
-                requestBody.setAttachments(getAttachments());
+            List<String> usersSentTo = new ArrayList<String>();
+            boolean sendToUser;
+
+            //send to default channel if enabled
+            if(sendDefaultChannel){
+                sendToChannel(this.getChannel(), requestBody, url);
             }
 
-            String bodyParam = String.format("payload=%s", URLEncoder.encode(requestBody.toJson(), UTF8));
+            if(sendUsers){
+                String slackUser;
+                for(Commit commit : commits){
+                    if(commit.hasSlackUserId()){
+                        sendToUser = true;
+                        slackUser = commit.getSlackUserId();
+
+                        /*Skip sending to the user if the commit was a merge from someone else
```

Comment:
The first part of the comment should be close to the code it is talking about. The merge commit check happens after this conditional.